### PR TITLE
Add python-pulp-cli-deb 0.0.2 to rpm/3.16

### LINKED ISSUE #322: Package pulp-cli-deb
Hi

Is it possible to package https://pypi.org/project/pulp-cli-deb/.

Wanted to create a pull request but can't find on how to create them for a pull request.

Is it possible to package this?

### PR DESCRIPTION
**Pull request by quba42**
Based on the following original PR: https://github.com/theforeman/pulpcore-packaging/pull/324

Closes #322

Applies some of my suggestions from the other PR, and bumps the version to 0.0.2.

### CODE REVIEW

**Comment by quba42** [comment]
I am still unsure about this: https://github.com/theforeman/pulpcore-packaging/pull/324#discussion_r775490529
and this: https://github.com/theforeman/pulpcore-packaging/pull/324#discussion_r777512135

File: comps/comps-pulpcore-el8.xml [hunk 1]
```diff
@@ -139,6 +139,7 @@
       <packagereq type="default">python38-pulp-ansible</packagereq>
       <packagereq type="default">python38-pulp-certguard</packagereq>
       <packagereq type="default">python38-pulp-cli</packagereq>
+      <packagereq type="default">python38-pulp-cli-deb</packagereq>
```
Comment (quba42) [inline comps/comps-pulpcore-el8.xml:142]: Not 100% sure what this does or if it is needed. (or if there are additional files like this one, the package should be added to).

**Comment by quba42** [comment]
The original PR had the following at the top of the SPEC file:
```
%{?scl:%scl_package python-%{pypi_name}}
%{!?scl:%global pkg_name %{name}}
```
I got rid of this, but I am not 100% sure about that, so please tell me if it is needed.

File: comps/comps-pulpcore-el8.xml (hunk 1 above)
Comment (evgeni) [reply comps/comps-pulpcore-el8.xml:142]: a similar change is needed to `comps-pulpcore-el7.xml`

**Comment by evgeni** [comment]
> The original PR had the following at the top of the SPEC file:
> 
> ```
> %{?scl:%scl_package python-%{pypi_name}}
> %{!?scl:%global pkg_name %{name}}
> ```
> 
> I got rid of this, but I am not 100% sure about that, so please tell me if it is needed.

it is needed, as now you've built the package *outside* the SCL (so against python 3.6, not 3.8)

**Comment by evgeni** [comment]
Also, can you rebase against `rpm/3.16`, nothing is using the 3.15 branch anymore.

**Comment by evgeni** [comment]
FTR, installing the built RPM on EL8 with `pulp-cli` from the 3.16 repo (`python38-pulp-cli-0.12.0-1.el8.noarch`) seems to work like a charm, `pulp --help` lists a `deb` command and `pulp deb --help` also works.

File: comps/comps-pulpcore-el8.xml (hunk 1 above)
Comment (quba42) [reply comps/comps-pulpcore-el8.xml:142]: Done.

**Comment by quba42** [comment]
> Also, can you rebase against `rpm/3.16`, nothing is using the 3.15 branch anymore.

Done. The repo's default branch on GitHub should probably be changed to `rpm/3.16` then.

**Comment by evgeni** [comment]
> > Also, can you rebase against `rpm/3.16`, nothing is using the 3.15 branch anymore.
> 
> Done. The repo's default branch on GitHub should probably be changed to `rpm/3.16` then.

i've just did exactly that :)

**Review by evgeni — APPROVED** [review]

**Comment by evgeni** [comment]
```console
% koji add-pkg --owner kojiadm pulpcore-3.16-el7 tfm-pulpcore-python-pulp-cli-deb                                                                                            
Adding 1 packages to tag pulpcore-3.16-el7
% koji add-pkg --owner kojiadm pulpcore-3.16-el8 python-pulp-cli-deb            
Adding 1 packages to tag pulpcore-3.16-el8
```

**Comment by evgeni** [comment]
Thanks @quba42 and @maartenbeeckmans!

**Pull request merged**